waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -135,15 +149,32 @@ dbSeed genProjectDir seedName = do
     ExitSuccess -> Right ()
     ExitFailure c -> Left $ "Failed with exit code " <> show c
 
-isDbRunning ::
+testDbConnection ::
   Path' Abs (Dir ProjectRootDir) ->
-  IO Bool
-isDbRunning genProjectDir = do
+  IO DbConnectionTestResult
+testDbConnection genProjectDir = do
   chan <- newChan
   exitCode <- DbJobs.dbExecuteTest genProjectDir chan
-  -- NOTE: We only care if the command succeeds or fails, so we don't look at
-  -- the exit code or stdout/stderr for the process.
-  return $ exitCode == ExitSuccess
+
+  case exitCode of
+    ExitSuccess -> return DbConnectionSuccess
+    ExitFailure _ -> do
+      outputLines <- collectJobTextOutputUntilExitReceived chan
+      let databaseNotCreated = any containsDatabaseNotCreatedError outputLines
+
+      return $
+        if databaseNotCreated
+          then DbNotCreated
+          else DbConnectionFailure
+
+-- Prisma error code for "Database not created" is P1003.
+containsDatabaseNotCreatedError :: T.Text -> Bool
+containsDatabaseNotCreatedError text = text TR.=~ ("\\bP1003\\b" :: String)
```

Comment:
I would probably put this in `where` under the `testDbConnection`. But maybe you took it out so you can write tests for it? If so, I understand. But then I would give it a bit more descriptive name: e.g. `prismaOutputContainsDbNotCreatedError`, because now it is missing the context it would have if it was in the `where`.